services/com.mde-network.ep.toolfunctions.xtext/src/main/java/com/mdenetnetwork/ep/toolfunctions/xtext/XtextTool.java

```diff
@@ -69,18 +69,32 @@ public void run(String languageName, String baseName, String extension, String g
 		
 		CliProjectsCreatorMain.main(args);
 		
-		
-		
-		
 		// Add activity files
-		if (grammar != null) {
-			final String grammarSrcPath = languageName.replace('.', '/');
-			final String xtextGrammarPath= PROJECT_PATH + baseName + "/src/" + grammarSrcPath  + ".xtext";
-			recreateGrammarFile(xtextGrammarPath, grammar);
+        //     TODO support multiple file types
+        final String grammarPath = PROJECT_PATH + baseName + "/src/" + baseName.replace('.', '/');
+        final String projectSrcPath = Path.of(grammarPath).getParent().toString();
```

Comment:
`projectSrcPath` is a confusing variable name here: it suggests this is `PROJECT_PATH + baseName + "/src/"`, when, really, it's just the path one level up from where the grammar lives.

I'm also not sure why it needs to be one level up? In a normal Xtext project, the validator, generator, and scope provider live in a sub-folder of the folder with the `.xtext` file... See my comments below.